### DISCUSSION TITLE
Trouble loading DLL into PowerShell?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [LibreHardwareMonitor/LibreHardwareMonitor#1211] [General]
Trouble loading DLL into PowerShell?

Add-Type -LiteralPath "C:\Automation\Library\LibreHardwareMonitor\LibreHardwareMonitorLib.dll"

Add-Type : Could not load file or assembly 'file:///C:\Automation\Library\LibreHardwareMonitor\LibreHardwareMonitorLib.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
At C:\Automation\Test.ps1:7 char:1
\+ Add-Type -LiteralPath "C:\Automation\Library\LibreHardwareMonitor\Lib ...
\+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
\+ CategoryInfo          : NotSpecified: (:) [Add-Type], FileLoadException
\+ FullyQualifiedErrorId : System.IO.FileLoadException,Microsoft.PowerShell.Commands.AddTypeCommand
 

Hit Line breakpoint on 'C:\Automation\Test.ps1:10'

**Comment by Jibun-no-Kage** [comment]
After unblocking, using .NET LoadFile... worked.  Using LoadModule worked.  Just odd that Add-Type sees to fail.  Still testing.

**Comment by jason-grimme** [comment]
This may help: https://github.com/Lifailon/PowerShell.HardwareMonitor
